### PR TITLE
fix KT-8032 Mixed platform library in JVM module causes JS warning ab…

### PR DESCRIPTION
**Pull request by MichaelNedzelsky**
…out unsupported format

#KT-8032 Fixed

### CODE REVIEW

**Comment by MichaelNedzelsky** [comment]
@udalov
Please take a look at this commit

File: idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java [hunk 1]
```diff
@@ -63,12 +66,36 @@ private KotlinRuntimeLibraryUtil() {}
...
+                             public boolean apply(@Nullable Module module) {
+                                 return module != null && ProjectStructureUtil.isJavaKotlinModule(module);
+                             }
+                            },
+                            new Predicate<Integer>() {
+                             @Override
+                             public boolean apply(Integer abiVersion) {
+                                 return !AbiVersionUtil.isAbiVersionCompatible(abiVersion);
+                             }
+                            });
```
Comment (udalov) [inline idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java]: Something wrong with formatting I suppose?

File: idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java [hunk 2]
```diff
@@ -287,16 +314,24 @@ static void replaceFile(File updatedFile, VirtualFile replacedJarFile) {
...
+        Module[] modules = ModuleManager.getInstance(project).getModules();
+
+        List<Module> modulesToCheck = new ArrayList<Module>();
+        for(Module module : modules) {
+            if (checkModule.apply(module)) {
+                modulesToCheck.add(module);
             }
-        }));
+        }
+        if (modulesToCheck.isEmpty()) return new ArrayList<VirtualFile>();
```
Comment (udalov) [inline idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java]: `Collections.emptyList()`?

File: idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java (hunk 1 above)
Comment (MichaelNedzelsky) [inline idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java]: In really, I don't know what type of format should be here.

File: idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java (hunk 2 above)
Comment (MichaelNedzelsky) [inline idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java]: Agreed, I'll fix it (and `UnsupportedAbiVersionNotificationPanelProvider.collectBadRoots`, where some files added to result of this method).

File: idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java [hunk 3]
```diff
@@ -287,16 +314,24 @@ static void replaceFile(File updatedFile, VirtualFile replacedJarFile) {
...
-        Collection<Integer> abiVersions = FileBasedIndex.getInstance().getAllKeys(id, project);
-        Set<Integer> badAbiVersions = Sets.newHashSet(Collections2.filter(abiVersions, new Predicate<Integer>() {
-            @Override
-            public boolean apply(Integer abiVersion) {
-                return !AbiVersionUtil.isAbiVersionCompatible(abiVersion);
+
+        Module[] modules = ModuleManager.getInstance(project).getModules();
+
+        List<Module> modulesToCheck = new ArrayList<Module>();
+        for(Module module : modules) {
```
Comment (udalov) [inline idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java]: If you're using `Predicate`, you might as well go ahead and call `Collections2.filter` here. Also I would recommend to use `kotlin.Function1` and `KotlinPackage.filter` instead, but that's up to you

File: idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java (hunk 1 above)
Comment (udalov) [inline idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java]: Just add a newline between `(` and `project` and press Ctrl+Shift+L / Cmd+Shift+L

**Comment by udalov** [comment]
It's ok, I don't see any problems, but what have you tested after these changes?

**Comment by MichaelNedzelsky** [comment]
I created Kotlin/Jvm project and added shared library (which contains both .class files and .js files). This library have valid JVM ABI but invalid JS ABI. In that case there is no notification now.

**Comment by MichaelNedzelsky** [comment]
@udalov 
fixes
https://github.com/JetBrains/kotlin/commit/2f43c56e5e71239ce3ebb2aea1280c74a5296d05

**Comment by MichaelNedzelsky** [comment]
Merged manually.

**Comment by udalov** [comment]
Ok, thanks

**Pull request closed without merging**